### PR TITLE
MWPW 142643

### PR DESCRIPTION
**Pull request by VKniaz**
3-Digit Quantity Support: Updated the quantity input handler to accept values up to three digits;
Max-Input Attribute: Implemented a 'max-input' attribute for the quantity input field, enabling a dynamic limit on the maximum value a user can input.
Default-value Attribute: Implemented a 'default-value' attribute for the quantity select default option.
Enhanced User Experience: Added a "+" symbol to the last selectable option in the quantity dropdown to indicate to users that higher quantities are available through direct input.
Auto-Focus on Last Option: Engineered the dropdown so that clicking on the last option with the "+" symbol automatically focuses on the input field, facilitating a seamless transition for users to enter their desired quantity.

Resolves: [mwpw-142643](https://jira.corp.adobe.com/browse/mwpw-142643)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/vkniaziev/qunatity-select?martech=off
- After: https://mwpw-142643--milo--vkniaz.hlx.page/drafts/vkniaziev/qunatity-select?martech=off

### CODE REVIEW

**Review by 3ch023 — APPROVED** [review]
Approved, but 2 things to address:

- merge conflicts with Stage branch
- this PR is touching the merch-card which is live on Production, so it doesn't sound like 0 impact, please ask either Reuban or Mili to QA it.

**Comment by VKniaz** [comment]
@3ch023 conflicts resolved.

**Comment by VKniaz** [comment]
@Roycethan, could you please verify this? I know you have already verified this in Tacocat PR, but still.

File: libs/blocks/merch-card/merch-card.js [hunk 1]
```diff
@@ -198,11 +198,13 @@ async function extractQuantitySelect(el) {
...
-    .filter((value) => /^\d+$/.test(value));
-  if (quantityValues.length !== 3) return null;
+  const values = config[1].textContent.split(',')
+    .map((value) => value.trim())
+    .filter((value) => /^\d*$/.test(value))
+    .map((value) => (value === '' ? undefined : Number(value)));
+  if (![3, 4, 5].includes(values.length)) return null;
   await import('../../deps/merch-quantity-select.js');
-  [attributes.min, attributes.max, attributes.step] = quantityValues.map(Number);
+  [attributes.min, attributes.max, attributes.step, attributes['default-value'], attributes['max-input']] = values;
```
Comment (Axelcureno) [inline libs/blocks/merch-card/merch-card.js:207]: This changes doesn't appear on https://git.corp.adobe.com/wcms/tacocat.js/pull/543/files I'm curious where this change is in tacocat?

**Review by Axelcureno — APPROVED** [review]

**Comment by Roycethan** [comment]
@VKniaz Plz update the Milo docs for authoring guidelines

**Pull request merged**